contrib/rust/src/python/pants/contrib/rust/tasks/cargo_toolchain.py

```diff
@@ -0,0 +1,141 @@
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import distutils.spawn
+import hashlib
+import os
+
+import requests
+from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TaskError
+from pants.base.workunit import WorkUnit, WorkUnitLabel
+from pants.util.process_handler import subprocess
+
+from pants.contrib.rust.tasks.cargo_task import CargoTask
+
+
+class Toolchain(CargoTask):
+  LAST_KNOWN_FINGERPRINT = '40328ad8fa5cfc15cdb0446bb812a4bba4c22b5aee195cfb8d64b8ef1de5879c'
+
+  @classmethod
+  def supports_passthru_args(cls):
+    return True
+
+  @classmethod
+  def register_options(cls, register):
+    super(Toolchain, cls).register_options(register)
+    register(
+      '--toolchain',
+      type=str,
+      default=None,
+      help='Toolchain')
+
+  @classmethod
+  def product_types(cls):
+    return ['cargo_env']
+
+  @classmethod
+  def implementation_version(cls):
+    return super(Toolchain, cls).implementation_version() + [('Cargo_Toolchain', 1)]
+
+  def setup_rustup(self):
+    self.context.log.info('Installing rustup...')
+    install_script = self.download_rustup_install_script()
+    self.check_integrity_of_rustup_install_script(install_script.text)
+
+    with self.context.new_workunit(name='setup-rustup',
+                                   labels=[WorkUnitLabel.BOOTSTRAP]) as workunit:
+      cmd = 'curl https://sh.rustup.rs -sSf | sh -s -- -y'  # -y Disable confirmation prompt.
+      self.run_shell_command(cmd, workunit)
+
+      if workunit.outcome() != WorkUnit.SUCCESS:
+        self.context.log.error(workunit.outcome_string(workunit.outcome()))
+      else:
+        self.context.log.info(workunit.outcome_string(workunit.outcome()))
+
+  def run_shell_command(self, command, workunit):
+    std_out = workunit.output('stdout')
+    std_err = workunit.output('stderr')
+
+    try:
+      subprocess.check_call(command, shell=True, stdout=std_out, stderr=std_err)
+    except subprocess.CalledProcessError as e:
+      workunit.set_outcome(1)
+      std_err.write('Execution failed: {0}'.format(e))
+    workunit.set_outcome(3)
+
+  def download_rustup_install_script(self):
+    return requests.get('https://sh.rustup.rs')
+
+  def check_integrity_of_rustup_install_script(self, install_script):
+    current_fingerprint = hashlib.sha256(install_script).hexdigest()
+    if current_fingerprint != self.LAST_KNOWN_FINGERPRINT:
+      raise TaskError(
+        'The fingerprint of the rustup script has changed!\nLast known: {0}\ncurrent: {1}'.format(
+          self.LAST_KNOWN_FINGERPRINT, current_fingerprint))
+
+  def install_rust_toolchain(self, toolchain):
+    self.context.log.info('Installing toolchain: {0}'.format(toolchain))
+    with self.context.new_workunit(name='install-rustup-toolchain',
+                                   labels=[WorkUnitLabel.BOOTSTRAP]) as workunit:
+      cmd = ['rustup', 'install', toolchain]
+
+      env = dict({
```

Comment:
You can drop the call to `dict()` and use the dict literal you have embedded into the function call. They are the same, and the latter is simpler.

```python
>>> dict({"test": 1}) == {"test": 1}
True
```

There are several places in this PR you can remove the call `dict()`.